pxtlib/package.ts

```diff
@@ -567,12 +570,32 @@ namespace pxt {
                 initPromise = initPromise.then(() => this.parseConfig(str))
             }
 
-            if (this.level == 0)
-                initPromise = initPromise.then(() => this.upgradePackagesAsync())
+            // if we are installing this script, we haven't yet downloaded the config
+            // do upgrade later
+            if (this.level == 0 && !isInstall) {
+                initPromise = initPromise.then(() => this.upgradePackagesAsync().then(() => { }))
+            }
 
             if (isInstall)
                 initPromise = initPromise.then(() => this.downloadAsync())
 
+            // we are installing the script, and we've download the original version and we haven't upgraded it yet
+            // do upgrade and reload as needed
+            if (this.level == 0 && isInstall) {
+                initPromise = initPromise.then(() => this.upgradePackagesAsync())
+                    .then(fixes => {
+                        if (fixes) {
+                            // worst case scenario with double load
+                            pxt.tickEvent("package.doubleload", fixes);
```

Comment:
This tickEvent needs careful review. Does it add tons of indices? Like versions. Then we need to remove the fixes array. Can you provide me a sample

Reply:
You can split into multiple events in a loop. Something like "extension": "xyz", "version": 1.0.1 , "upgraded version": 2.05

Reply:
adding one tick per extension